File: app/src/main/java/com/bukarev/camerahelper/MainActivity.java
package com.bukarev.camerahelper;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;

import java.io.File;

public class MainActivity extends AppCompatActivity implements OnCameraResult {
    private CameraHelper cameraHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        cameraHelper = CameraHelper.newBuilder(this, 1, 2, 3, 4)
                .setErrorInterface(this)
                .build();
        cameraHelper.startCameraWithPermission(true, false);
        cameraHelper.startPickingFromGallery();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        cameraHelper.onSaveInstanceState(outState);
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        cameraHelper.onRestoreInstanceState(savedInstanceState);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        File file = cameraHelper.onActivityResult(requestCode, resultCode, data);
        String qwe = "";
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        cameraHelper.onPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    public void onCameraError(Throwable error) {

    }
}

File: app/src/main/java/com/robotboy/camerahelper/app/MainActivity.kt
package com.robotboy.camerahelper.app

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.davemorrissey.labs.subscaleview.ImageSource
import com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView
import com.robotboy.camerahelper.helper.CameraErrorInterface
import com.robotboy.camerahelper.helper.CameraHelper
import com.robotboy.camerahelper.R

class MainActivity : AppCompatActivity(), CameraErrorInterface {
    lateinit var cameraHelper: CameraHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val cameraBtn = findViewById<Button>(R.id.camera)
        cameraHelper = CameraHelper.newBuilder(this, 1, 2, 3, 4).setErrorInterface(this).build()
        cameraBtn.setOnClickListener {
            cameraHelper.startCameraWithPermission(false)
        }
    }

    override fun onCameraError(error: Throwable?) {
        // TODO show error
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 1 && resultCode == Activity.RESULT_OK) {
            val imageFile = cameraHelper.onActivityResult(requestCode, resultCode, data)
            val imageView = findViewById<SubsamplingScaleImageView>(R.id.image)
            imageView.setImage(ImageSource.uri(cameraHelper.currentUri))
        }
    }
}
